lib/ansible/module_utils/mlnxos.py

```diff
@@ -0,0 +1,77 @@
+# -*- coding: utf-8 -*-
+#
+# (c) 2017, Ansible by Red Hat, inc
+#
+# This file is part of Ansible by Red Hat
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+from ansible.module_utils._text import to_text
+from ansible.module_utils.basic import env_fallback
+from ansible.module_utils.connection import exec_command
+from ansible.module_utils.network_common import to_list, ComplexList
+
+
+_DEVICE_CONFIGS = {}
+
+mlnxos_provider_spec = {
+    'host': dict(),
+    'port': dict(type='int'),
+    'username': dict(fallback=(env_fallback,
+                               ['ANSIBLE_NET_USERNAME'])),
+    'password': dict(fallback=(env_fallback,
+                               ['ANSIBLE_NET_PASSWORD']), no_log=True),
+    'ssh_keyfile': dict(fallback=(env_fallback,
+                                  ['ANSIBLE_NET_SSH_KEYFILE']), type='path'),
+    'authorize': dict(fallback=(env_fallback,
+                                ['ANSIBLE_NET_AUTHORIZE']), type='bool'),
+    'auth_pass': dict(fallback=(env_fallback,
+                                ['ANSIBLE_NET_AUTH_PASS']), no_log=True),
+    'timeout': dict(type='int')
+}
+mlnxos_argument_spec = {
+    'provider': dict(type='dict', options=mlnxos_provider_spec),
+}
+
+
+def get_provider_argspec():
+    return mlnxos_provider_spec
+
+
+def check_args(module, warnings):
+    pass
+
+
+def to_commands(module, commands):
+    spec = {
+        'command': dict(key=True),
+        'prompt': dict(),
+        'answer': dict()
+    }
+    transform = ComplexList(spec, module)
+    return transform(commands)
+
+
+def run_commands(module, commands, check_rc=True):
+    responses = list()
+    commands = to_commands(module, to_list(commands))
+    for cmd in commands:
+        cmd = module.jsonify(cmd)
+        rc, out, err = exec_command(module, cmd)
```

Comment:
From 2.5 onwards network platforms using persistent connection will be using cliconf and netconf plugins. To use cliconf plugin you need to add platform-specific plugin in `lib/ansible/plugins/cliconf/<platform-file>`
You can refer `asa` modules as it is implemented using cliconf api's
https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/asa.py
https://github.com/ansible/ansible/blob/devel/lib/ansible/plugins/cliconf/asa.py